go/vt/servenv/status.go

```diff
@@ -34,41 +34,46 @@ import (
 	"vitess.io/vitess/go/vt/log"
 )
 
-var (
-	binaryName  = filepath.Base(os.Args[0])
-	hostname    string
-	serverStart = time.Now()
-
-	statusMu       sync.RWMutex
-	statusSections []section
-	statusTmpl     = template.Must(reparse(nil))
-	statusFuncMap  = make(template.FuncMap)
-)
-
-type section struct {
-	Banner   string
-	Fragment string
-	F        func() interface{}
+// AddStatusPart adds a new section to status. frag is used as a
+// subtemplate of the template used to render /debug/status, and will
+// be executed using the value of invoking f at the time of the
+// /debug/status request. frag is parsed and executed with the
+// html/template package. Functions registered with AddStatusFuncs
+// may be used in the template.
+func AddStatusPart(banner, frag string, f func() interface{}) {
```

Comment:
nit - I would rename `frag`, to be  `fragment`. I spent a bit of time thinking how hand grenades relate to all these templates. :rofl: